### ISSUE TITLE
cart subtotal shipping value it free, while it should not

### ISSUE THREAD

**Issue by Flower7C3** [PrestaShop/PrestaShop#12775]
**Describe the bug**
Shipping cost is always free for not logged person.

**To Reproduce**
1. Add any non virtual item to cart.
2. Go to cart page. In summary box shipping is free - which is invalid (screenshot 1).
3. Modify file `classes/Cart.php` in line 3233 add anything (screenshot 2).
4. Reload cart page. In summary box shipping is not free - which is OK (screenshot 3).
5. Reload cart page again. In summary box shipping is free again - which is invalid.

**Screenshots**

1. <img width="212" alt="zrzut ekranu 2019-03-2 o 22 21 05" src="https://user-images.githubusercontent.com/3767861/53687830-fa1ac780-3d39-11e9-8723-1df7d396d6cb.png">
2. <img width="426" alt="zrzut ekranu 2019-03-2 o 22 20 16" src="https://user-images.githubusercontent.com/3767861/53687802-6812bf00-3d39-11e9-83d7-81e564ef2612.png">
3. <img width="213" alt="zrzut ekranu 2019-03-2 o 22 21 45" src="https://user-images.githubusercontent.com/3767861/53687831-fd15b800-3d39-11e9-9cd6-4dc2747d8020.png">

**Additionnal information**
PrestaShop version: 1.7.5.0
PHP version: 7.3

**Comment by Flower7C3** [comment]
Probably the problem is with math equations from 3229 and 3230 lines in `classes/Cart.php` file:
`$_total_shipping['with_tax'] += $delivery_option_list[$id_address][$key]['total_price_with_tax'];
$_total_shipping['without_tax'] += $delivery_option_list[$id_address][$key]['total_price_without_tax'];`

Everything is ok when I change it to:
`$_total_shipping['with_tax'] = $_total_shipping['with_tax'] + $delivery_option_list[$id_address][$key]['total_price_with_tax'];
$_total_shipping['without_tax'] = $_total_shipping['without_tax'] + $delivery_option_list[$id_address][$key]['total_price_without_tax'];`


So for now the fix is to override this class in `override/classes/Cart.php` file like this:
 
```php
<?php

class Cart extends CartCore
{
    public function getTotalShippingCost($delivery_option = null, $use_tax = true, Country $default_country = null)
    {
        if (isset(Context::getContext()->cookie->id_country)) {
            $default_country = new Country(Context::getContext()->cookie->id_country);
        }
        if (is_null($delivery_option)) {
            $delivery_option = $this->getDeliveryOption($default_country, false, false);
        }

        $_total_shipping = array(
            'with_tax' => 0,
            'without_tax' => 0,
        );
        $delivery_option_list = $this->getDeliveryOptionList($default_country);

        foreach ($delivery_option as $id_address => $key) {
            if (!isset($delivery_option_list[$id_address]) || !isset($delivery_option_list[$id_address][$key])) {
                continue;
            }

$_total_shipping['with_tax'] = $_total_shipping['with_tax'] + $delivery_option_list[$id_address][$key]['total_price_with_tax'];
$_total_shipping['without_tax'] = $_total_shipping['without_tax'] + $delivery_option_list[$id_address][$key]['total_price_without_tax'];
        }

        return ($use_tax) ? $_total_shipping['with_tax'] : $_total_shipping['without_tax'];
    }
}
```

**Comment by khouloudbelguith** [comment]
Hi @Flower7C3,

In your BO => IMPROVE => Carriers => Preferences page, you can edit the carrier by default, so in the FO, you can have a carrier with not a free value.
I attached a video record.
https://drive.google.com/file/d/1M7KN8jHOhCnHIkQKwUd0vL3NF301OuaQ/view
Thanks to check & feedback.

**Comment by khouloudbelguith** [comment]
@Flower7C3, PrestaShop is not compatible with PHP 7.3, in fact, PrestaShop 1.7.0 to 1.7.3 is compatible with PHP 7.1 and PrestaShop 1.7.4 & 1.7.5 are compatible with PHP 7.2.
For more details, you can follow this link: http://build.prestashop.com/news/prestashop-1-7-is-moving-to-symfony-3-4-and-php-5-6/
Thanks

**Comment by Flower7C3** [comment]
@khouloudbelguith in BO default carrier is set. I have checked it on clean setup with PHP 7.2 - problem also occurs.
Please note, that it could be fixed when You change this calculation:
```
$_total_shipping['with_tax'] += $delivery_option_list[$id_address][$key]['total_price_with_tax'];
$_total_shipping['without_tax'] += $delivery_option_list[$id_address][$key]['total_price_without_tax'];
```
with longer version:
```
$_total_shipping['with_tax'] = $_total_shipping['with_tax'] + $delivery_option_list[$id_address][$key]['total_price_with_tax'];
$_total_shipping['without_tax'] = $_total_shipping['without_tax'] + $delivery_option_list[$id_address][$key]['total_price_without_tax'];
```

**Comment by khouloudbelguith** [comment]
@Flower7C3, the carrier by default in your case is free shipping or not?
Why, when you have free shipping, is invalid?
Thanks!

**Comment by Flower7C3** [comment]
@khouloudbelguith by default I have non free shipping carrier, but in cart there if free shipping message. As I wrote there is a bug with shipping cost calculation.

**Comment by khouloudbelguith** [comment]
@Flower7C3, but in my case, with fresh installation 1.7.5.1, I did not manage to reproduce the issue.
https://drive.google.com/file/d/1fkY98VvLoVtAwSJyBrO2z1G99WggTRtj/view
Could you please provide us with more info? We need more details to understand how we can reproduce your issue:
 - server setup and configuration
 - debug mode report
 - PHP error logs
 - apache error log
 - javascript console log
 - screenshots

Don't you know how to get this information? Please read the following article:
http://build.prestashop.com/howtos/misc/how-to-create-bug-report/

Thanks!

**Comment by rblaurin** [comment]
@khouloudbelguith, I can reproduce It. Try to add a new product to the cart in a new incognito window and then go to the cart. 
It seems that when no delivery info is provided yet, Prestashop assumes it is a Free shipping, but in this case no shipping prices would be shown.

**Comment by khouloudbelguith** [comment]
Hi @rblaurin,

Thanks for your feedback.
I tried with PS1.7.5.1 but it is OK.
https://drive.google.com/file/d/12TZgo09WkiXScmgcUJ-AWYzNCMSGlrtG/view
Thanks!

**Comment by rblaurin** [comment]
Hi, @khouloudbelguith

It occurs for me when the shipping value changes according to the delivery address. I think it is not the case in your configuration because the shipping value is shown before you select your country.

**Comment by khouloudbelguith** [comment]
@rblaurin, here's my carrier setting
![image](https://user-images.githubusercontent.com/16067358/53802871-3bca8e80-3f43-11e9-914d-e076fe649bb0.png)

I defined it as my carrier by default
I added a new product to the cart in a new incognito window and then I went to the cart.
I have this
![image](https://user-images.githubusercontent.com/16067358/53802933-76ccc200-3f43-11e9-8bf9-22fd9abce2f3.png)

Thanks!

**Comment by rblaurin** [comment]
Is It the right behavior? You did't enter your address, so you could be in Africa, right? Don't you think the shipping cost should be shown only after you had selected your country?

**Comment by Flower7C3** [comment]
@khouloudbelguith I use Prestashop 1.7.5.0 on PHP 7.3. Mabye it works in 1.7.5.1 and different PHP version.

**Comment by rblaurin** [comment]
@Flower7C3 could you share your carrier setting? I have a similar problem with 1.7.5.1 and php 7.2.

**Comment by Flower7C3** [comment]
@rblaurin do You mean all data from AdminCarriers page or AdminCarrierPrefferences?

**Comment by rblaurin** [comment]
@Flower7C3, both, if you can.

**Comment by Flower7C3** [comment]
@rblaurin in prefferences I have selected **Best price**. My customer has configured 6 carriers. For query `select * from ps_carrier join ps_delivery on ps_carrier.id_carrier = ps_delivery.id_carrier where ps_carrier.deleted = 0` I have following result:

id_carrier | id_reference | id_tax_rules_group | name | url | active | deleted | shipping_handling | range_behavior | is_module | is_free | shipping_external | need_range | external_module_name | shipping_method | position | max_width | max_height | max_depth | max_weight | grade | id_delivery | id_shop | id_shop_group | id_carrier | id_range_price | id_range_weight | id_zone | price
------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------ | ------------
103 | 41 | 0 | Kurier DPD |  | 1 | 0 | 0 | 1 | 0 | 0 | 0 | 0 |  | 1 | 0 | 0 | 0 | 0 | 0.000000 | 0 | 163 | NULL | NULL | 103 | NULL | 15 | 1 | 11.300000
104 | 19 | 0 | Kurier DPD pobranie |  | 1 | 0 | 0 | 1 | 0 | 0 | 0 | 0 |  | 1 | 1 | 0 | 0 | 0 | 0.000000 | 0 | 164 | NULL | NULL | 104 | NULL | 14 | 1 | 12.930000
105 | 5 | 0 | Paczkomat Inpost |  | 1 | 0 | 0 | 1 | 0 | 0 | 0 | 0 |  | 1 | 2 | 0 | 0 | 0 | 0.000000 | 0 | 165 | NULL | NULL | 105 | NULL | 16 | 1 | 10.160000
106 | 98 | 0 | Paczkomat pobranie |  | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |  | 1 | 3 | 0 | 0 | 0 | 0.000000 | 0 | 166 | NULL | NULL | 106 | NULL | 17 | 1 | 11.790000
108 | 43 | 0 | Dostawa pocztą |  | 1 | 0 | 0 | 1 | 0 | 0 | 0 | 0 |  | 1 | 5 | 0 | 0 | 0 | 0.000000 | 0 | 168 | NULL | NULL | 108 | NULL | 4 | 1 | 0.000000
109 | 1 | 0 | Odbiór z magazynu |  | 1 | 0 | 0 | 1 | 0 | 0 | 0 | 0 |  | 1 | 4 | 0 | 0 | 0 | 0.000000 | 0 | 169 | NULL | NULL | 109 | NULL | 8 | 1 | 0.000000

**Comment by khouloudbelguith** [comment]
Hi @Flower7C3, @rblaurin,

I manage to reproduce the issue with the last version 1.7.5.1
Steps to reproduce the issue:
1- there are no carriers in the shop
![image](https://user-images.githubusercontent.com/16067358/47090058-5dd8a400-d222-11e8-8875-aff40d7c8f91.png)
2- Navigate to FO
3- Add some products to the cart => process to checkout
4- In the third step: an alert is displayed"Unfortunately, there are no carriers available for your delivery address." but in the cart summary: The shipping label is displayed with free value.
![image](https://user-images.githubusercontent.com/16067358/47090221-c0ca3b00-d222-11e8-9020-b0e6b3a78ba0.png)
We are aware of this issue, it is added to our debug roadmap.
Your ticket is a duplicate of #10739
So, I close it.
@Flower7C3, would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop
Thank you!

**Comment by khouloudbelguith** [comment]
Duplicate of #10739
